docs/spec/spec.md

```diff
@@ -0,0 +1,556 @@
+it’s being transferred. This means data is not exposed to outside entities on the Internet when it is sent
+to another party.
+
+Ballerina FTP library contains two core apis:
+* Client - The `ftp:Client` is used to connect to an FTP server and perform various operations on the files.
+* Listener - The `ftp:Listener` is used to listen to a remote FTP location and notify if files are added or removed 
+  from the FTP location.
+
+## 2. Configurations
+### 2.1. Security Configurations
+* PrivateKey record represents the configuration related to a private key.
+```ballerina
+public type PrivateKey record {|
+    # Path to the private key file
+    string path;
+    # Private key password
+    string password?;
+|};
+```
+* Credentials record represents the username and password configurations.
+```ballerina
+public type Credentials record {|
+    # Username of the user
+    string username;
+    # Password of the user
+    string password;
+|};
+```
+* AuthConfiguration record represents the configurations needed for facilitating secure communication with a remote FTP server.
+```ballerina
+public type AuthConfiguration record {|
+    # Username and password to be used
+    Credentials credentials?;
+    # Private key to be used
+    PrivateKey privateKey?;
+|};
+```
+### 2.2. FileInfo
+* `FileInfo` record contains the metadata of the files.
+```ballerina
+public type FileInfo record {|
+    # Relative file path for a newly-added file
+    string path;
+    # Size of the file
+    int size;
+    # Last-modified timestamp of the file in UNIX Epoch time
+    int lastModifiedTimestamp;
+    # File name
+    string name;
+    # `true` if the file is a folder
+    boolean isFolder;
+    # `true` if the file is a file
+    boolean isFile;
+    # Normalized absolute path of this file within its file system
+    string pathDecoded;
+    # Extension of the file name
+    string extension;
+    # Receiver as a URI String for public display
+    string publicURIString;
+    # Type of the file
+    string fileType;
+    # `true` if the `fileObject` is attached
+    boolean isAttached;
+    # `true` if someone reads/writes from/to this file
+    boolean isContentOpen;
+    # `true` if this file is executable
+    boolean isExecutable;
+    # `true` if this file is hidden
+    boolean isHidden;
+    # `true` if this file can be read
+    boolean isReadable;
+    # `true` if this file can be written
+    boolean isWritable;
+    # Depth of the file name within its file system
+    int depth;
+    # URI scheme of the file
+    string scheme;
+    # Absolute URI of the file
+    string uri;
+    # Root URI of the file system in which the file exists
+    string rootURI;
+    # A "friendly path" is a path, which can be accessed without a password
+    string friendlyURI;
+|};
+```
+## 3. Client
+The `ftp:Client` connects to an FTP server and performs various operations on the files. Currently, it supports the
```

Comment:
```suggestion
The `ftp:Client` connects to FTP server and performs various operations on the files. Currently, it supports the
```

Reply:
Here also article is not needed.

Reply:
Updated in https://github.com/ballerina-platform/module-ballerina-ftp/pull/538/commits/3df62c276651402b0d11a31cc5968282ed4cba5f